packages/status-js/src/utils/encode-url-data.test.ts

```diff
@@ -0,0 +1,92 @@
+import { describe, expect, test } from 'vitest'
+
+import {
+  decodeChannelUrlData,
+  decodeCommunityUrlData,
+  decodeUserUrlData,
+  encodeChannelUrlData,
+  encodeCommunityUrlData,
+  encodeUserUrlData,
+} from './encode-url-data'
+
+describe('URL data', () => {
+  test('should encode and decode community', () => {
+    const data = {
+      displayName: 'Lorem ipsum dolor sit egestas.',
+      description:
+        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus non dui vitae augue elementum laoreet ac pharetra odio. Morbi vestibulum.',
+      membersCount: 1_000_000,
+      color: '#4360DF',
+    }
+    const publicKey = 'zQ3shY7r4cAdg4eUF5dfcuCqCFzWmdjHW4SX5hspM9ucAarfU'
+
+    const encodedData = encodeCommunityUrlData(data, publicKey)
+    const decodedData = decodeCommunityUrlData(encodedData, publicKey)
+
+    expect(encodedData).toBe(
+      'G8QAgB2HsXukxPjOtKWIubcNl-B0uURsnKDo9aJV-0VzRbRGSUCoqm5G0xr2x3HvncEPqJXanmOcKGy6yMLbtw8espSvaBJj6QDp8OCCzq2V8uEqwVGdSfACIGwxf7i4WCazAtZ1dF_tAU5lm3De56yL7Ru9hE9phNUowsGzFqZ5kAl7cf4WzDB4lzebHo_w8aaempKoyas6hpxKbJ3d_wY='
+    )
+    expect(decodedData).toEqual(data)
+  })
+
+  test('should throw on invalid checksum for community encoded with incorrect public key', () => {
```

Comment:
ℹ️ checksum use case

Reply:
can you elaborate, please?

Reply:
This demonstrates one way of preventing

- _you_ creating a new community/profile with offending content (e.g. name, description)
- copying the generated share link
- slapping _my/mine_ public key at the end
- and telling everybody I'm the author of the content

However, if you knew my public key before hand and spent a little effort by generating the link programmatically, this wouldn't catch it. And that is also the reason why I think this checksum is even worth – can't assume the data is valid.